app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -7,17 +7,30 @@ module AuthnJwt
     Authenticator = CommandClass.new(
       dependencies: {
         token_factory: TokenFactory.new,
-        logger: Rails.logger
+        logger: Rails.logger,
+        audit_log: ::Audit.logger,
+        validate_origin: ::Authentication::ValidateOrigin.new,
+        role_class: ::Role,
+        webservice_class: ::Authentication::Webservice,
+        validate_role_can_access_webservice: ::Authentication::Security::ValidateRoleCanAccessWebservice.new,
       },
       inputs: %i[jwt_configuration authenticator_input]
     ) do
+      extend(Forwardable)
+      def_delegators(:@authenticator_input, :account, :username, :client_ip, :authenticator_name, :service_id)
 
       def call
```

Comment:
where do we validate that the webservice is whitelisted?

Reply:
In @tzheleznyak 's [PR](https://github.com/cyberark/conjur/pull/2203).
My branch actually is based on his one.
All "bug fixes" are temporary in there, Tamir is handling it as a part of his PR.